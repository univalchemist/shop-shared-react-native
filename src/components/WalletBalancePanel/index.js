import React from 'react';
import PropTypes from 'prop-types';
import { isNil } from 'ramda';
import {
  TextSkeletonPlaceholder,
  Flex,
  Box,
  FormattedMoney,
  Text,
  SecondaryText,
} from '@wrappers/components';
import { FormattedMessage } from 'react-intl';

const Panel = props => (
  <Flex
    justifyContent="space-between"
    flexDirection="row"
    px={4}
    py={3}
    borderBottomWidth={1}
    borderColor="gray.5"
    bg="white"
    {...props}
  />
);

const Label = props => (
  <SecondaryText fontSize={14} lineHeight={16} fontWeight="bold" {...props} />
);

const WalletBalancePanelLayout = ({ text, balance, ...props }) => (
  <Panel>
    <Box>
      <Flex flexDirection="row">
        <Label>{text ? text : <FormattedMessage id="wallet" />}</Label>
      </Flex>
    </Box>
    <Box>{balance}</Box>
  </Panel>
);

const WalletBalancePanel = ({ balance, text }) => (
  <WalletBalancePanelLayout
    text={text}
    balance={
      <Text fontSize={14} lineHeight={16} fontWeight="bold">
        {!isNil(balance) && <FormattedMoney value={balance} />}
      </Text>
    }
  />
);

WalletBalancePanel.propTypes = {
  balance: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

const WalletBalanceSkeletonPlaceholder = () => (
  <WalletBalancePanelLayout
    text={<FormattedMessage id="wallet" />}
    balance={
      <TextSkeletonPlaceholder fontSize={14} lineHeight={16} width={128} />
    }
  />
);

const WalletBalancePanelError = () => (
  <Panel>
    <Label>
      <FormattedMessage id="wallet.error" />
    </Label>
  </Panel>
);

export { WalletBalancePanelError, WalletBalanceSkeletonPlaceholder };
export default WalletBalancePanel;
