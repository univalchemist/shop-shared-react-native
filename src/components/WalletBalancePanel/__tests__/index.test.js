import { renderForTest } from '@testUtils';
import React from 'react';
import WalletBalancePanel from '../index';

describe('WalletBalancePanel', () => {
  it('should render correctly with balance', () => {
    const component = renderForTest(<WalletBalancePanel balance={1000} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render correctly with text', () => {
    const component = renderForTest(
      <WalletBalancePanel text={'Not available'} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
