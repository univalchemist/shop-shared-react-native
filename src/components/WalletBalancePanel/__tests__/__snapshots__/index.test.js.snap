// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletBalancePanel should render correctly with balance 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        bg="white"
                        borderBottomWidth={1}
                        borderColor="gray.5"
                        flexDirection="row"
                        justifyContent="space-between"
                        px={4}
                        py={3}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#ffffff",
                              "borderColor": "#e3e3e3",
                              "display": "flex",
                              "flexDirection": "row",
                              "flexWrap": "wrap",
                              "justifyContent": "space-between",
                              "paddingBottom": 16,
                              "paddingLeft": 32,
                              "paddingRight": 32,
                              "paddingTop": 16,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {},
                            ]
                          }
                        >
                          <View
                            flexDirection="row"
                            style={
                              Array [
                                Object {
                                  "display": "flex",
                                  "flexDirection": "row",
                                  "flexWrap": "wrap",
                                },
                              ]
                            }
                          >
                            <Text
                              color="gray.3"
                              fontFamily="default"
                              fontSize={14}
                              fontWeight="bold"
                              lineHeight={16}
                              style={
                                Array [
                                  Object {
                                    "color": "#9a9a9a",
                                    "fontFamily": "Arial",
                                    "fontSize": 14,
                                    "fontWeight": "700",
                                    "lineHeight": 16,
                                  },
                                ]
                              }
                            >
                              <Text>
                                Wellness balance
                              </Text>
                            </Text>
                          </View>
                        </View>
                        <View
                          style={
                            Array [
                              Object {},
                            ]
                          }
                        >
                          <Text
                            color="gray.1"
                            fontFamily="default"
                            fontSize={14}
                            fontWeight="bold"
                            lineHeight={16}
                            style={
                              Array [
                                Object {
                                  "color": "#646464",
                                  "fontFamily": "Arial",
                                  "fontSize": 14,
                                  "fontWeight": "700",
                                  "lineHeight": 16,
                                },
                              ]
                            }
                          >
                            <Text>
                              HK$1,000.00
                            </Text>
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;

exports[`WalletBalancePanel should render correctly with text 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        bg="white"
                        borderBottomWidth={1}
                        borderColor="gray.5"
                        flexDirection="row"
                        justifyContent="space-between"
                        px={4}
                        py={3}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#ffffff",
                              "borderColor": "#e3e3e3",
                              "display": "flex",
                              "flexDirection": "row",
                              "flexWrap": "wrap",
                              "justifyContent": "space-between",
                              "paddingBottom": 16,
                              "paddingLeft": 32,
                              "paddingRight": 32,
                              "paddingTop": 16,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {},
                            ]
                          }
                        >
                          <View
                            flexDirection="row"
                            style={
                              Array [
                                Object {
                                  "display": "flex",
                                  "flexDirection": "row",
                                  "flexWrap": "wrap",
                                },
                              ]
                            }
                          >
                            <Text
                              color="gray.3"
                              fontFamily="default"
                              fontSize={14}
                              fontWeight="bold"
                              lineHeight={16}
                              style={
                                Array [
                                  Object {
                                    "color": "#9a9a9a",
                                    "fontFamily": "Arial",
                                    "fontSize": 14,
                                    "fontWeight": "700",
                                    "lineHeight": 16,
                                  },
                                ]
                              }
                            >
                              Not available
                            </Text>
                          </View>
                        </View>
                        <View
                          style={
                            Array [
                              Object {},
                            ]
                          }
                        >
                          <Text
                            color="gray.1"
                            fontFamily="default"
                            fontSize={14}
                            fontWeight="bold"
                            lineHeight={16}
                            style={
                              Array [
                                Object {
                                  "color": "#646464",
                                  "fontFamily": "Arial",
                                  "fontSize": 14,
                                  "fontWeight": "700",
                                  "lineHeight": 16,
                                },
                              ]
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
